build.gradle

```diff
@@ -1,71 +1,115 @@
+        }
+    }
     jcenter()
     mavenCentral()
-    maven { url "https://dl.bintray.com/kotlin/kotlinx" }
-    maven { url "https://jitpack.io" }
 }
 
 dependencies {
-    // Kotlin
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 
-    // Cloud Events
-    compile 'com.github.open-broker:cloud-events-jvm:b8b9c6fef37'
-    compile 'com.github.open-broker:cloud-events-jvm:b8b9c6fef37:sources'
+    implementation(
+        platform("org.jetbrains.kotlin:kotlin-bom:$ver.kotlin"),
+        platform("com.fasterxml.jackson:jackson-bom:$ver.jackson"),
+        "org.jetbrains.kotlin:kotlin-stdlib-jdk8",
+        "com.fasterxml.jackson.core:jackson-core",
+        "com.fasterxml.jackson.module:jackson-module-kotlin",
+        "com.fasterxml.jackson.datatype:jackson-datatype-jsr310",
+        "com.github.open-broker:cloud-events-jvm:$ver.cloudEvents",
+    )
 
-    // Jackson
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version:'2.9.8'
+    testImplementation(
+        platform("org.junit:junit-bom:$ver.junit"),
+        "org.junit.jupiter:junit-jupiter-api",
+    )
 
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.3.1")
-    testRuntime("org.junit.jupiter:junit-jupiter-engine:5.3.1")
+    testRuntime(
+        "org.junit.jupiter:junit-jupiter-engine",
+    )
 }
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
     kotlinOptions {
-        languageVersion = kotlin_api_version
-        apiVersion = kotlin_api_version
-        jvmTarget = jvm_version
-        javaParameters = true   // Useful for reflection.
+        javaParameters = true
+        jvmTarget = jdkVersion
+        freeCompilerArgs += ['-Xjsr305=strict', '-progressive']
+        languageVersion = kotlinLanguageVersion
+        apiVersion = kotlinLanguageVersion
     }
 }
 
-compileJava {
-    sourceCompatibility = jvm_version
-    targetCompatibility = jvm_version
-    options.incremental = true
-    options.encoding = 'UTF-8'
+tasks.withType(JavaCompile).all {
+    options.encoding = defaultEncoding
+    options.compilerArgs = ['-parameters', '-Xlint:unchecked']
+}
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    archiveClassifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+        }
+    }
 }
 
 test {
+
     useJUnitPlatform()
-}
 
-task wrapper(type: Wrapper) {
-    description = 'Generates gradlew[.bat] scripts for faster execution'
-    gradleVersion = '4.10.2'
+    reports.junitXml.enabled false
+    reports.html.enabled false
+
+    failFast = System.getProperty('failFast') != null
```

Comment:
This is not really nice. I know its configurable but its not really needed when all test take just 1-2 sec to run. Better to run all tests.